seamless-immutable.development.js

```diff
@@ -326,20 +326,24 @@
     return result;
   }
 
+  // Creates empty object, using prototype if present
+  function instantiateEmptyObject() {
+    var prototype = Object.getPrototypeOf(this);
+    return prototype !== Object.prototype ? Object.create(prototype) : {};
+  }
+
   // Finalizes an object with immutable methods, freezes it, and returns it.
   function makeImmutableObject(obj) {
     addPropertyTo(obj, "merge", merge);
     addPropertyTo(obj, "without", without);
     addPropertyTo(obj, "asMutable", asMutableObject);
+    addPropertyTo(obj, "instantiateEmptyObject", instantiateEmptyObject);
```

Comment:
+1 for passing it as an argument to `makeImmutableObject`! That sounds like the best approach yet.